cmd/telemetry_provider/telemetry_provider.go

```diff
@@ -0,0 +1,381 @@
+	Long:  "Update Telemetry Provider",
+	Run: func(cmd *cobra.Command, args []string) {
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		authApi.GetInfo("", "")
+
+		telemetryProviderName, _ := cmd.Flags().GetString("config-name")
+		sinkType, _ := cmd.Flags().GetString("type")
+
+		sinkTypeEnum, err := ybmclient.NewTelemetryProviderTypeEnumFromValue(strings.ToUpper(sinkType))
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		oldname := telemetryProviderName
+		if cmd.Flags().Changed("new-config-name") {
+			telemetryProviderName, _ = cmd.Flags().GetString("new-config-name")
+		}
+		//We initialise this one here, even if we error out later
+		telemetryProviderSpec, err := setTelemetryProviderConfiguration(cmd, telemetryProviderName, *sinkTypeEnum)
+		if err != nil {
+			logrus.Fatalf(err.Error())
+		}
+
+		config, err := authApi.GetConfigByName(oldname)
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		resp, r, err := authApi.UpdateTelemetryProvider(config.GetInfo().Id).TelemetryProviderSpec(*telemetryProviderSpec).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		msg := fmt.Sprintf("The telemetryProvider %s is being updated", formatter.Colorize(config.GetInfo().Id, formatter.GREEN_COLOR))
+
+		fmt.Println(msg)
+
+		telemetryProviderCtx := formatter.Context{
+			Output: os.Stdout,
+			Format: formatter.NewTelemetryProviderFormat(viper.GetString("output"), string(resp.GetData().Spec.Type)),
+		}
+
+		respArr := []ybmclient.TelemetryProviderData{resp.GetData()}
+
+		formatter.TelemetryProviderWrite(telemetryProviderCtx, respArr)
+	},
+}
+
+var validateTelemetryProviderCmd = &cobra.Command{
+	Use:   "validate",
+	Short: "Validate Telemetry Provider",
+	Long:  "Validate Telemetry Provider",
+	Run: func(cmd *cobra.Command, args []string) {
+
+		telemetryProviderName, _ := cmd.Flags().GetString("config-name")
+		sinkType, _ := cmd.Flags().GetString("type")
+
+		//We initialise here, even if we error out later
+		sinkTypeEnum, err := ybmclient.NewTelemetryProviderTypeEnumFromValue(strings.ToUpper(sinkType))
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		telemetryProviderSpec, err := setTelemetryProviderConfiguration(cmd, telemetryProviderName, *sinkTypeEnum)
+		if err != nil {
+			logrus.Fatalf(err.Error())
+		}
+		authApi, err := ybmAuthClient.NewAuthApiClient()
+		if err != nil {
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+		authApi.GetInfo("", "")
+
+		resp, r, err := authApi.ValidateTelemetryProviderConfig().TelemetryProviderSpec(*telemetryProviderSpec).Execute()
+
+		if err != nil {
+			logrus.Debugf("Full HTTP response: %v", r)
+			logrus.Fatalf(ybmAuthClient.GetApiErrorDetails(err))
+		}
+
+		isValid := resp.GetData().IsValid
+
+		if isValid {
```

Comment:
Why do we need a separate validation ? Can we just not vaidate when we create or update?

Reply:
We have an api to validate telemetry provider config in ybm-backend and all the generated SDKs have that too.
Hence, added an api in CLI as well. 
We do the validations on create/update and return a failed response if the telemetry config is not valid.

CLI use case for this api could be checking a list of configs present with user and select from a valid set based on some prioritisation logic, or maybe some pre-compute step to avoid failing the CRUD operations.

Reply:
Not sure I see any value on this valid function ... 
Validation should be done transparently.

Reply:
removed validation